대응 표본 T-검정 (Paired T-test)
표본의 각 사례마다 대응하는 2개의 관측치를 통해 판단

한 집단에 어떤 작용이 가해졌을 때에 대한 before, after를 비교

In [1]:
from scipy import stats
from pandas import read_excel

#### 예제 (1) - 사슴의 다리 길이 비교
사슴의 왼쪽 뒷다리와 왼쪽 앞다리의 길이를 측정한 자료를 통해 두 다리의 길이(cm)가 같은지 다른지 알고 싶다.

이 경우 한 사슴에 대하여 두 다리 길이를 측정하였으므로 사슴은 독립이지만 한 마리의 사슴의 일부인 두 다리는 독립이 아니다

In [2]:
df = read_excel("https://data.hossam.kr/E02/deer.xlsx", index_col="사슴")
df

뒷다리  앞다리
사슴          
1   142  138
2   140  136
3   144  147
4   144  139
5   142  143
6   146  141
7   149  143
8   150  145
9   142  136
10  148  146

#### Case 1

|가설|내용|
|---|---|
|귀무가설|사슴의 앞다리와 뒷다리 길이에는 차이가 없다. |
|대립가설|사슴의 앞다리와 뒷다리 길이에는 차이가 존재한다.|

In [3]:
stats.ttest_rel(df['앞다리'], df['뒷다리'], alternative='two-sided')


TtestResult(statistic=-3.413793103448276, pvalue=0.007703223347263766, df=9)

#### 결과 해석

차이값에 대한 t 통계량의 값이 `3.41`, 대응되는 유의확률이 `0.0077`로 0.05보다 작으므로 유의수준 0.05에서 앞다리와 뒷다리의 길이는 같지 않다고 결론 내릴 수 있다.



#### Case 2

|가설|내용|
|---|---|
|귀무가설|사슴의 앞다리와 뒷다리 길이에는 길지 않다. |
|대립가설|사슴의 앞다리와 뒷다리 길이에는 길다|

In [4]:
stats.ttest_rel(df['앞다리'], df['뒷다리'], alternative='greater')


TtestResult(statistic=-3.413793103448276, pvalue=0.9961483883263681, df=9)

#### 결과 해석

차이값에 대한 t 통계량의 값이 `3.41`, 대응되는 유의확률이 `0.99`로 0.05보다 크므로 유의수준 0.05에서 앞다리는 뒷다리보다 길지 않다고 결론 내릴 수 있다.

#### Case 3

|가설|내용|
|---|---|
|귀무가설|사슴의 앞다리와 뒷다리 길이에는 짧지 않다. |
|대립가설|사슴의 앞다리와 뒷다리 길이에는 짧다.|

In [5]:
stats.ttest_rel(df['앞다리'], df['뒷다리'], alternative='less')

TtestResult(statistic=-3.413793103448276, pvalue=0.003851611673631883, df=9)

#### 결과 해석

차이값에 대한 t 통계량의 값이 `3.41`, 대응되는 유의확률이 `0.0038`로 0.05보다 작으므로 유의수준 0.05에서 앞다리는 뒷다리보다 짧다고 결론 내릴 수 있다.